experimental/auth-logic/wrapper_interface.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package authlogic
+
+import (
+	"fmt"
+	"os"
+)
+
+// This struct represents an authorization logic statement (or sequence of
+// statements) without a principal. These statements might include rules or
+// predicates. For example, the following is an unattributed statement:
+// ```
+// foo("bar").
+// "SomePrincipal" canActAs baz("bar") :- foo("bar").
+// ```
+// NOTE: Because wrappers will generally emit UnattributedStatements,
+// it might be useful to also give a field for the relation declarations
+// for the relations referenced in the statement. Without this, the
+// interface will work by having the code that consumes the wrappers emit
+// all the necessary declarations for the wrappers it uses.
+type UnattributedStatement struct {
+	Contents string
+}
+
+func (statement UnattributedStatement) String() string {
+	return statement.Contents
+}
+
+type Principal struct {
+	Contents string
+}
+
+func (principal Principal) String() string {
+	return principal.Contents
+}
+
+type AuthLogic struct {
```

Comment:
Is this a good name? Should not it be something like AuthLogicPredicate or something?